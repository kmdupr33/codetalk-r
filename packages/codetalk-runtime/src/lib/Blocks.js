import React from "react";
import PropTypes from "prop-types";
import SyntaxHighlighter from "react-syntax-highlighter";

const Blocks = ({ input }) => {
  console.log(`Blocks: ${JSON.stringify(input)}`);
  // Blocks: [["#","Hello world"]]
  // Blocks: [["##"," goodbye"]]
  // Blocks: ["idk why you say goodbye"]
  const [block, emptyBlocks] = input;
  const getRenderedBlock = () => {
    if (!Array.isArray(block)) {
      return <p>{input.join("")}</p>;
    }
    if (block[0] === "#") {
      return <h1>{block[1]}</h1>;
    }
    if (block[0] === "##") {
      return <h2>{block[1]}</h2>;
    }
    if (block[0] === "```") {
      const [_, language, code] = block;
      return <SyntaxHighlighter language={language}>{code}</SyntaxHighlighter>;
    }
    return input;
  };
  return (
    <React.Fragment>
      {getRenderedBlock()}
      {emptyBlocks && emptyBlocks.map(_ => <br/>)}
    </React.Fragment>
  );
};

Blocks.propTypes = {
  input: PropTypes.array.isRequired
};

export default Blocks;
