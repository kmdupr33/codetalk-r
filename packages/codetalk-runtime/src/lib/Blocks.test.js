import React from "react";
import Blocks from "./Blocks";
import renderer from "react-test-renderer";

describe("Blocks", () => {
  it("handles a simple title", () => {
    const tree = renderer
      .create(<Blocks input={[["#", "Hello world"]]} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("handles a simple subtitle", () => {
    const tree = renderer.create(<Blocks input={[["##", "goodbye"]]} />);
    expect(tree).toMatchSnapshot();
  });
  it("handles simple text", () => {
    const tree = renderer.create(<Blocks input={["idk why you say goodbye"]} />);
    expect(tree).toMatchSnapshot();
  })  
  it("handles a newline", () => {
    
  });
});
