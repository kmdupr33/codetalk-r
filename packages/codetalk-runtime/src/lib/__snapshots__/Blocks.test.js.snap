// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blocks handles a simple subtitle 1`] = `
<h2>
  goodbye
</h2>
`;

exports[`Blocks handles a simple title 1`] = `
<h1>
  Hello world
</h1>
`;

exports[`Blocks handles simple text 1`] = `
<p>
  idk why you say goodbye
</p>
`;
